quantus/metrics/base.py

```diff
@@ -287,14 +243,12 @@ def __call__(
     @abstractmethod
     def evaluate_instance(
         self,
-        i: int,
+        i: int,  # TODO: remove this from the general case and check why we need this workaround.
```

Comment:
Not sure what is meant by the TODO comment?

Reply:
i have now removed this argument from the default arguments, as there are only two metrics which use this.
we have this new `custom_preprocess()` functionality and I used it for adding `i_batch` and the rest of the attributes.

Reply:
Agree on that solution!